self_unenrollment_tool/views.py

```diff
@@ -0,0 +1,245 @@
+                logger.info(f'deleted self enrollment for user {se_user_id}, course_instance {se_course_instance_id}, role {se_role_id}', extra=extra)
+                successes.append('Successfully deleted self-enrollment')
+            except Exception as e:
+                logger.error(f'Failed to delete self-enrollment for user {user_sis_id} with role {se_role_id} in course {course_sis_id}', extra=extra)
+                errors.append('Failed to delete self-enrollment')
+
+            for ce in canvas_enrollments:
+                if ce['role_id'] == canvas_role_id:
+                    # we need to delete this one
+                    try:
+                        result = conclude_enrollment(request_ctx=SDK_CONTEXT, course_id=ce['course_id'], id=ce['id'], task='delete')
+                        if result.status_code == 200:
+                            logger.info(f'deleted Canvas enrollment id {ce["id"]} in course {ce["course_id"]}', extra=extra)
+                            successes.append('Successfully deleted Canvas self-enrollment')
+                        else:
+                            logger.error(f'Failed to delete Canvas self-enrollment for user {user_sis_id} with role {canvas_role_id} in course {course_sis_id}', extra=extra)
+                            errors.append('Failed to delete Canvas self-enrollment')
+                    except Exception as e:
+                        logger.exception(f'Failed to delete Canvas self-enrollment for user {user_sis_id} with role {canvas_role_id} in course {course_sis_id}', extra=extra)
+                        errors.append('Feled to delete Canvas self-enrollment')
+
+
+        message = ''
+        if errors:
+            if successes:
+                # partially successful
+                message = 'Removing your self-enrollment(s) for this course was partially successful. If you remain enrolled after 24 hours, check with your teaching staff.'
+            else:
+                # not successful
+                message = 'Removing your self-enrollments(s) for this course was not successful. Please check with your teaching staff.'
+
+            messages.add_message(request, messages.ERROR, message)
+            return redirect(reverse('self_unenrollment_tool:error'))
+
+        return redirect(reverse('self_unenrollment_tool:success', kwargs={'launch_id': launch_id}))
+
+
+def _get_self_enrollments(course_sis_id: str, user_sis_id: str):
+    # this function fetches all three types of enrollment and returns them in one list
+    enrollments = []
+    staff_enrollments = CourseStaff.objects.filter(course_instance_id=course_sis_id, user_id=user_sis_id, source='selfenroll').all()
+    student_enrollments = CourseEnrollee.objects.filter(course_instance_id=course_sis_id, user_id=user_sis_id, source='selfenroll').all()
+    guest_enrollments = CourseGuest.objects.filter(course_instance_id=course_sis_id, user_id=user_sis_id, source='selfenroll').all()
+    for e in staff_enrollments:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)

Similar issue at line numbers 170 and 172.

Reply:
Not sure I agree re: list comprehensions always being easier to understand :-)